packages/builder/src/pages/builder/app/[application]/design/[screenId]/[componentId]/_components/Component/ComponentSettingsSection.svelte

```diff
@@ -19,6 +19,8 @@
   export let includeHidden = false
   export let tag
 
+  $: store.actions.components.processNestedSettings(componentInstance)
```

Comment:
Is this to try and do the migration just whenever the component is rendered for the first time? I reckon it might be cleaner to move this logic actually into the `selectedComponent` derived store itself, right before it returns the instance - which would mean that by using `$selectedComponent` anywhere we already have the up to date migrated instance. It also escapes any race conditions which could happen here since there will be an initial render of the settings section before this reactive statement runs.

Do you think it might also make sense to name this `migrateSettings` as well? To make it clear that the intention is to migrate legacy settings into new ones.